sdks/java/extensions/sql/datacatalog/src/main/java/org/apache/beam/sdk/extensions/sql/meta/provider/datacatalog/DataCatalogClientAdapter.java

```diff
@@ -31,63 +30,51 @@
 import io.grpc.ManagedChannelBuilder;
 import io.grpc.MethodDescriptor;
 import io.grpc.auth.MoreCallCredentials;
-import java.io.IOException;
+import java.util.function.Supplier;
 import javax.annotation.Nullable;
+import org.apache.beam.sdk.extensions.gcp.options.GcpOptions;
 import org.apache.beam.sdk.extensions.sql.meta.Table;
 
 /** Wraps DataCatalog GRPC client and exposes simplified APIS for Data Catalog Table Provider. */
 class DataCatalogClientAdapter {
 
-  private DataCatalogBlockingStub dcClient;
+  private DataCatalogPipelineOptions options;
 
-  private DataCatalogClientAdapter(DataCatalogBlockingStub dcClient) {
-    this.dcClient = dcClient;
+  private Supplier<DataCatalogBlockingStub> dcClient = () -> newClient(options);
+
+  private DataCatalogClientAdapter(DataCatalogPipelineOptions options) {
+    this.options = options;
   }
 
   /** Prod endpoint (default set in pipeline options): datacatalog.googleapis.com. */
-  public static DataCatalogClientAdapter withDefaultCredentials(String endpoint)
-      throws IOException {
-    return new DataCatalogClientAdapter(newClient(endpoint));
+  public static DataCatalogClientAdapter withOptions(DataCatalogPipelineOptions options) {
+    return new DataCatalogClientAdapter(options);
   }
 
-  private static DataCatalogBlockingStub newClient(String endpoint) throws IOException {
+  private DataCatalogBlockingStub newClient(DataCatalogPipelineOptions options) {
     Channel authedChannel =
         ClientInterceptors.intercept(
-            ManagedChannelBuilder.forTarget(endpoint).build(),
-            CredentialsInterceptor.defaultCredentials());
+            ManagedChannelBuilder.forTarget(options.getDataCatalogEndpoint()).build(),
+            new CredentialsInterceptor());
     return DataCatalogGrpc.newBlockingStub(authedChannel);
```

Comment:
You don't need to create a CredentialsInterceptor class, you can invoke `withCallCredentials(creds)` on the return from `newBlockingStub`

Reply:
Nice.